app/src/main/java/unipd/se18/ocrcamera/PhotoTester.java

```diff
@@ -0,0 +1,306 @@
+            Log.e(TAG, "Error writing report to file.");
+            e.printStackTrace();
+        }
+        return fullReport;
+    }
+
+
+    /**
+     * Write the string given as argument to a file named with the fileName argument
+     * @param report text that will be saved to filename
+     * @param dirPath path to a directory with writing permissions
+     * @param fileName name of the file - will be overwritten if already exist
+     * @author Luca Moroldo (g3)
+     * @throws IOException
+     */
+    private void writeReportToExternalStorage(String report, String dirPath, String fileName) throws IOException {
+        //Write report to report.txt
+        File file = new File(dirPath, fileName);
+
+        file.createNewFile();
+        FileOutputStream stream = null;
+        try {
+            stream = new FileOutputStream(file);
+            try {
+                stream.write(report.getBytes());
+            } finally {
+                stream.close();
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            Log.e(TAG, "File not found");
+        }
+    }
+
+
+    /**
+     * Apply OCR to the given bitmap
+     * @param bitmap from which the text is extracted
+     * @return String - the text extracted
+     */
+    private String executeOcr(Bitmap bitmap) {
+        TextExtractor textExtractor = new TextExtractor();
+        return textExtractor.getTextFromImg(bitmap);
+    }
+
+
+    /**
+     * Class used to evaluate a single TestElement
+     * @author Luca Moroldo (g3)
+     */
+    private class RunnableTest implements Runnable {
+        private TestElement test;
+        private JSONObject jsonReport;
+        private CountDownLatch countDownLatch;
+
+        /**
+         * @param jsonReport JSONObject where will be added the evaluation of the TestElement
+         * @param test - TestElement that must be associated to a bitmap
+         * @param countDownLatch used to signal the task completion
+         */
+        public RunnableTest(JSONObject jsonReport, TestElement test, CountDownLatch countDownLatch) {
+            this.jsonReport = jsonReport;
+            this.test = test;
+            this.countDownLatch = countDownLatch;
+        }
+
+        @Override
+        public void run() {
+                //evaluate text extraction confidence
+                String extractedIngredients = executeOcr(test.getPicture());
+                String correctIngredients = test.getIngredients();
+                //compare the text recognized by the OCR with the real one
+                float confidence = ingredientsTextComparison(correctIngredients, extractedIngredients);
+
+                //insert results in test
+                test.setConfidence(confidence);
+                test.setRecognizedText(extractedIngredients);
+
+                //evaluate each alterations if present
+                String[] alterationsFileNames = test.getAlterationsNames();
```

Comment:
you want to test the difference between the words of ocr and the totals one??

Reply:
I want to evaluate the performance of the OCR comparing the recognized text with the real text, the evaluation is done also for any alteration of a photo. Do you think this wasn't clear?